shared/constants/devices.tsx

```diff
@@ -67,72 +69,28 @@ export const getEndangeredTLFs = (state: Container.TypedState, id?: Types.Device
 // Device -> [1, 10]
 // We split devices by type and order them by creation time. Then, we use (index mod 10)
 // as the background #
-const numBackgrounds = 10
-const idxMapper = (device: Types.Device, idx: number): [Types.DeviceID, number] => [
-  device.deviceID,
-  (idx % numBackgrounds) + 1,
-]
-type DeviceIconInfo = {
-  map: Map<Types.DeviceID, number>
-  next: {desktop: number; mobile: number}
-}
+export const numBackgrounds = 10
+
+type NextDeviceIconInfo = {desktop: number; mobile: number}
+
 const getIndexMap = memoize(
-  (devices: Map<Types.DeviceID, Types.Device>): DeviceIconInfo => {
-    const sorted = [...devices.values()]
-      .sort((a, b) => a.created - b.created)
-      .reduce<{
-        backup: Array<Types.Device>
-        desktop: Array<Types.Device>
-        mobile: Array<Types.Device>
-      }>(
-        (res, device) => {
-          switch (device.type) {
-            case 'backup':
-              res.backup.push(device)
-              break
-            case 'desktop':
-              res.desktop.push(device)
-              break
-            case 'mobile':
-              res.mobile.push(device)
-              break
-          }
-          return res
-        },
-        {backup: [], desktop: [], mobile: []}
-      )
-    return {
-      map: new Map([
-        ...sorted.backup.map(idxMapper),
-        ...sorted.desktop.map(idxMapper),
-        ...sorted.mobile.map(idxMapper),
-      ]),
-      next: {
-        desktop: (sorted.desktop.length % numBackgrounds) + 1,
-        mobile: (sorted.mobile.length % numBackgrounds) + 1,
-      },
-    }
+  (devices: Map<Types.DeviceID, Types.Device>): NextDeviceIconInfo => {
+    return [...devices.values()].reduce<NextDeviceIconInfo>(
```

Comment:
Can this find the max `deviceNumberOfType` for desktop/mobile and add one?

Reply:
Hmm it could.. that would be the incorrect icon if you've revoked your most recent device; that is probably ok though.